hepcrawl/spiders/osti_spider.py

```diff
@@ -0,0 +1,197 @@
+        """Constructs the URL to use for querying OSTI.
+
+        Returns:
+            str: URL
+        """
+        params = {}
+        params['entry_date_start'] = self.from_date \
+            or self.resume_from(set_=self.set)
+        if self.until_date:
+            params['entry_date_end'] = self.until_date
+        if self.rows:
+            params['rows'] = self.rows
+        if self.product_types:
+            params['product_type'] = self.product_types
+        if self.journal_types:
+            params['journal_type'] = self.journal_types
+        if self.research_org:
+            params['research_org'] = self.research_org
+
+        return furl(OSTISpider.osti_base_url).add(params).url
+
+    @property
+    def set(self):
+        """Acronym of the research organization for which to harvest publications.
+
+        Returns:
+            str: e.g. u'SLAC'
+        """
+        return self.research_org
+
+    @property
+    def headers(self):
+        """HTTP headers to use in requests.
+
+        Returns:
+            dict:
+        """
+        return {'Accept': 'application/json'}
+
+    def start_requests(self):
+        """Just yield the scrapy.Request."""
+        started_at = datetime.utcnow()
+        self.logger.debug("URL is %s" % self.url)
+        yield Request(self.url,
+                      headers=self.headers)
+        self.save_run(started_at=started_at, set_=self.set)
+
+    def parse(self, response):
+        """Parse OSTI response into HEP records.
+
+        Returns:
+            generator: dict: json
+        """
+
+        osti_response = json.loads(response.text)
+        for record in osti_response:
+            parser = OSTIParser(record, source=self.name)
+
+            yield ParsedItem(
+                record=parser.parse(),
+                record_format="hep",
+            )
+
+        total_count = response.headers.get('X-Total-Count')
+        if int(total_count) > len(osti_response):
+            self.logger.debug("do paging, total of %s records" % int(total_count))
+
+        # Pagination support. Will yield until no more "next" pages are found
+        if 'Link' in response.headers:
+            links = link_header.parse(response.headers[u'Link'].decode())
+            nextlink = links.links_by_attr_pairs([(u'rel', u'next')])
+            if nextlink:
+                next_url = nextlink[0].href
+                self.logger.debug("using next link %s" % next_url)
+                yield Request(next_url,
+                              headers=self.headers,
+                              callback=self.parse)
+
+    def make_file_fingerprint(self, set_):
+        """Create a label to use in filename storing last_run information.
+
+        Returns:
+            str:
+        """
+        return u'set={}'.format(set_)
+
+    @property
+    def product_types(self):
+        """Product type URL parameter
+
+        Returns:
+            str:
+        """
+        return self.__product_types
```

Comment:
one `_` is probably enough

Reply:
the `__` is intentional here, see e.g.  https://www.python-course.eu/python3_properties.php

because I want to enforce checks on setting